### PR TITLE
pkg/controller/statefulset/stateful_set_utils.go: Improving test coverage

### PR DESCRIPTION
**Pull request by Octopusjust**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improve test coverage

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Two high-level ideas:

- Prefer to test exported functions. For these functions, you could add more test cases to TestStatefulsetControl that cover more scenarios.
- Use test tables

**Comment by Octopusjust** [comment]
> Two high-level ideas:
> 
> * Prefer to test exported functions. For these functions, you could add more test cases to TestStatefulsetControl that cover more scenarios.
> * Use test tables

constructive suggestions for me, i will update it.

File: pkg/controller/statefulset/stateful_set_utils_test.go [hunk 1]
```diff
@@ -589,6 +589,101 @@ func TestIsRunningAndReady(t *testing.T) {
...
 
+func TestIsPodInRightStatus(t *testing.T) {
+	now := metav1.Now()
+	set := newStatefulSet(3)
+
+	testCases := []struct {
+		name            string
+		podPhase        v1.PodPhase
+		podCondition    v1.PodCondition
+		podDeletionTime metav1.Time
```
Comment (alculquicondor) [inline pkg/controller/statefulset/stateful_set_utils_test.go:600]:
```suggestion
		podDeletionTime *metav1.Time
```

**Comment by alculquicondor** [comment]
You are still testing non-exported functions. Furthermore, testing all of them in the same test table it's unconventional.

I prefer not to merge this unless you test an exported function.

**Comment by Octopusjust** [comment]
> 
Maybe I misunderstood your reply, I will revise it again, thanks

**Comment by Octopusjust** [comment]
> You are still testing non-exported functions. Furthermore, testing all of them in the same test table it's unconventional.
> 
> I prefer not to merge this unless you test an exported function.

I should have misunderstood your meaning. I will revise it again, thanks.

**Comment by Octopusjust** [comment]
> You are still testing non-exported functions. Furthermore, testing all of them in the same test table it's unconventional.
> 
> I prefer not to merge this unless you test an exported function.

Hi, I checked carefully again, these funcs have been covered by the testcases of exported functions in stateful_set_control.go. So there should be no need to add testcases again. Thanks for your review :)

**Pull request closed without merging**